src/main/kotlin/io/arrowkt/model/Account.kt

```diff
@@ -0,0 +1,151 @@
+package io.arrowkt.model
+
+import io.arrowkt.Amount
+import io.arrowkt.ErrorOr
+import io.arrowkt.ValidationResult
+import arrow.core.Option
+import arrow.core.getOrElse
+import arrow.core.none
+import arrow.core.some
+import arrow.data.NonEmptyList
+import arrow.data.extensions.nonemptylist.semigroup.semigroup
+import arrow.data.extensions.validated.applicative.applicative
+import arrow.data.fix
+import arrow.data.invalidNel
+import arrow.data.validNel
+import io.arrowkt.today
+import java.math.BigDecimal
+import java.time.LocalDate
+
+data class Balance(val amount: Amount = BigDecimal.ZERO)
+
+sealed class Account(
+    open val no: String,
+    open val name: String,
+    open val dateOfOpen: Option<LocalDate>,
+    open val dateOfClose: Option<LocalDate>,
+    open val balance: Balance
+) {
+    companion object {
```

Comment:
Since I can't have an object with the same name as the sealed class, I put this code in the Companion.

do you know of any alternatives?

Reply:
You could use a top-level object instead. Either way works.